peer/brontide.go

```diff
@@ -3804,18 +3804,28 @@ func (p *Brontide) chanFlushEventSentinel(chanCloser *chancloser.RbfChanCloser,
 	// We'll wait until the channel enters the ChannelFlushing state. We
 	// exit after a success loop. As after the first RBF iteration, the
 	// channel will always be flushed.
-	for newState := range newStateChan {
-		if _, ok := newState.(*chancloser.ChannelFlushing); ok {
-			peerLog.Infof("ChannelPoint(%v): rbf coop "+
-				"close is awaiting a flushed state, "+
-				"registering with link..., ",
-				channel.ChannelPoint())
-
-			// Request the link to send the event once the channel
-			// is flushed. We only need this event sent once, so we
-			// can exit now.
-			link.OnFlushedOnce(sendChanFlushed)
+	for {
+		select {
+		case newState, ok := <-newStateChan:
+			if !ok {
+				return
+			}
+
+			if _, ok := newState.(*chancloser.ChannelFlushing); ok {
+				peerLog.Infof("ChannelPoint(%v): rbf coop "+
+					"close is awaiting a flushed state, "+
+					"registering with link..., ",
+					channel.ChannelPoint())
 
+				// Request the link to send the event once the channel
+				// is flushed. We only need this event sent once, so we
+				// can exit now.
+				link.OnFlushedOnce(sendChanFlushed)
+
+				return
+			}
+
+		case <-p.cg.Done():
```

Comment:
Yep this was missing, as we launched the goroutine tied to the main peer scoped wait group: https://github.com/lightningnetwork/lnd/blob/cbf70f20493d5c9a5f80403b7dce80ed99e87bf9/peer/brontide.go#L3944-L3948

I think this may have been missed in itests, as once the main state machine is torn down, we'll close that channel which will unblock this goroutine: https://github.com/lightningnetwork/lnd/blob/3b2af90f5a7bde84244c7b137787d0a463b6998a/fn/conc_queue.go#L113-L114

Nice catch!